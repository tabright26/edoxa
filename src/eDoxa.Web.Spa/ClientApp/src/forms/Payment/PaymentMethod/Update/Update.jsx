import React from "react";
import { FormGroup, Form, Label } from "reactstrap";
import { Field, reduxForm, FormSection } from "redux-form";
import Button from "components/Shared/Override/Button";
import Input from "components/Shared/Override/Input";
import { UPDATE_PAYMENTMETHOD_FORM } from "forms";
import validate from "./validate";
import { months } from "utils/helper";
import CardExpirationMonthOption from "components/Payment/Card/Expiration/Month/Option";
import CardExpirationYearOption from "components/Payment/Card/Expiration/Year/Option";
import CardBrandIcon from "components/Payment/Card/BrandIcon";

const expYearOptions = exp_year => {
  const array = [];
  for (let index = exp_year; index < exp_year + 20; index++) {
    array.push(index);
  }
  return array;
};

const UpdatePaymentMethodForm = ({
  handleSubmit,
  initialValues: {
    card: { brand, last4, exp_year }
  }
}) => (
  <Form onSubmit={handleSubmit} inline className="d-flex">
    <FormGroup>
      <div className="d-flex">
        <CardBrandIcon className="my-auto" brand={brand} size="2x" />
        <span className="my-auto ml-2">{`XXXX XXXX XXXX ${last4}`}</span>
      </div>
    </FormGroup>
    <FormSection className="mx-auto" name="card" component={FormGroup}>
      <Label className="ml-4 mr-2 text-muted">Expiration:</Label>
      <Field className="d-inline" type="select" name="exp_month" style={{ width: "55px" }} component={Input.Select}>
        {months.map((month, index) => (
          <CardExpirationMonthOption key={index} month={month} />
        ))}
      </Field>
      <span className="d-inline mx-2">/</span>
      <Field className="d-inline" type="select" name="exp_year" style={{ width: "55px" }} component={Input.Select}>
        {expYearOptions(exp_year).map((year, index) => (
          <CardExpirationYearOption key={index} year={year} />
        ))}
      </Field>
    </FormSection>
    <Button.Save />
  </Form>
);

export default reduxForm({ form: UPDATE_PAYMENTMETHOD_FORM, validate })(UpdatePaymentMethodForm);
