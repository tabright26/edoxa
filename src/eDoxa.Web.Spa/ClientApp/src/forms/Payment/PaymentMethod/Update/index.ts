import UpdatePaymentMethodForm from "./Update";

export default UpdatePaymentMethodForm;
