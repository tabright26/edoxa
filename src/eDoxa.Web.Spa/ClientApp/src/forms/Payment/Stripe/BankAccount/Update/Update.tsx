import React, { FunctionComponent } from "react";
import Button from "components/Shared/Override/Button";
import Input from "components/Shared/Override/Input";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Form } from "reactstrap";
import { UPDATE_BANK_ACCOUNT_FORM } from "forms";
import { validate } from "./validate";
import { compose } from "recompose";

const UpdateStripeBankAccountForm: FunctionComponent<any> = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <Field type="text" name="currency" label="Currency" formGroup={FormGroup} component={Input.Text} />
    <Field type="text" name="accountHolderName" label="Account Holder Name" formGroup={FormGroup} component={Input.Text} />
    <Field type="text" name="routingNumber" label="Routing Number" formGroup={FormGroup} component={Input.Text} />
    <Field type="text" name="accountNumber" label="Account Number" formGroup={FormGroup} component={Input.Text} />
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

const enhance = compose<any, any>(reduxForm<any, { handleCancel: () => any }, string>({ form: UPDATE_BANK_ACCOUNT_FORM, validate }));

export default enhance(UpdateStripeBankAccountForm);
