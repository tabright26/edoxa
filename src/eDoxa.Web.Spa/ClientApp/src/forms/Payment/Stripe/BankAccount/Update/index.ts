import UpdateStripeBankAccountForm from "./Update";

export default UpdateStripeBankAccountForm;
