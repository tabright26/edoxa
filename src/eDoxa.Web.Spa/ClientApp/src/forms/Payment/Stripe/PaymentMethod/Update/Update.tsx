import React, { FunctionComponent } from "react";
import { FormGroup, Form, Label } from "reactstrap";
import { reduxForm, FormSection } from "redux-form";
import Button from "components/Shared/Override/Button";
import { UPDATE_STRIPE_PAYMENTMETHOD_FORM } from "forms";
import { validate } from "./validate";
import CardBrandIcon from "components/Payment/Card/BrandIcon";
import { compose } from "recompose";
import MonthSelectField from "components/Shared/Override/Field/Month";
import YearSelectField from "components/Shared/Override/Field/Year";

const UpdateStripePaymentMethodForm: FunctionComponent<any> = ({
  handleSubmit,
  initialValues: {
    card: { brand, last4, exp_year }
  },
  handleCancel,
  updateStripePaymentMethod
}) => (
  <Form onSubmit={handleSubmit((data: any) => updateStripePaymentMethod(data).then(() => handleCancel()))} inline className="d-flex">
    <FormGroup>
      <div className="d-flex">
        <CardBrandIcon className="my-auto" brand={brand} size="2x" />
        <span className="my-auto ml-2">{`XXXX XXXX XXXX ${last4}`}</span>
      </div>
    </FormGroup>
    <FormSection className="mx-auto" name="card">
      <Label className="ml-4 mr-2 text-muted">Expiration:</Label>
      <MonthSelectField className="d-inline" name="exp_month" width="55px" />
      <span className="d-inline mx-2">/</span>
      <YearSelectField className="d-inline" name="exp_year" width="55px" min={exp_year} max={exp_year + 20} descending={false} />
      {/* <Field className="d-inline" type="select" name="exp_year" style={{ width: "55px" }} component={Input.Select}>
        {expYearOptions(exp_year).map((year, index) => (
          <CardExpirationYearOption key={index} year={year} />
        ))}
      </Field> */}
    </FormSection>
    <Button.Save />
  </Form>
);

const enhance = compose<any, any>(reduxForm<any, { handleCancel: () => any }, string>({ form: UPDATE_STRIPE_PAYMENTMETHOD_FORM, validate }));

export default enhance(UpdateStripePaymentMethodForm);
