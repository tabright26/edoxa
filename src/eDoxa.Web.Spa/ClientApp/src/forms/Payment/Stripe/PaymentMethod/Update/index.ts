import UpdateStripePaymentMethodContainer from "./Container";

export default UpdateStripePaymentMethodContainer;
