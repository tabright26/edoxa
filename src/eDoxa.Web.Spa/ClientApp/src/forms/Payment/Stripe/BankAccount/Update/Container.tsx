import { connect } from "react-redux";
import { loadStripeBankAccount, updateStripeBankAccount } from "store/root/payment/stripe/bankAccount/actions";
import Update from "./Update";
import { compose } from "recompose";
import { injectStripe } from "react-stripe-elements";
import { withtUserProfile } from "store/root/user/container";

const mapDispatchToProps = (dispatch: any, ownProps: any) => {
  return {
    updateStripeBankAccount: (data: any) =>
      ownProps.stripe
        .createToken("bank_account", {
          account_holder_name: data.accountHolderName,
          routing_number: data.routingNumber,
          account_number: data.accountNumber,
          currency: data.currency,
          country: ownProps.country
        })
        .then(result => {
          if (result.token) {
            console.log(result.token);
            return dispatch(updateStripeBankAccount(result.token.id)).then(() => dispatch(loadStripeBankAccount()));
          } else {
            return Promise.reject(result.error);
          }
        })
  };
};

const enhance = compose<any, any>(
  injectStripe,
  withtUserProfile("country"),
  connect(
    null,
    mapDispatchToProps
  )
);

export default enhance(Update);
