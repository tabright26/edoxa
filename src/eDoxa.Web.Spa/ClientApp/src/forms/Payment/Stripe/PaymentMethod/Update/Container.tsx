import { connect } from "react-redux";
import { RootState } from "store/root/types";
import { loadStripePaymentMethods, updateStripePaymentMethod } from "store/root/payment/stripe/paymentMethods/actions";
import Update from "./Update";

const mapStateToProps = (state: RootState, ownProps: any) => {
  const {
    data: { data }
  } = state.payment.stripe.paymentMethods;
  const paymentMethod = data.find(paymentMethod => paymentMethod.id === ownProps.paymentMethodId);
  return {
    initialValues: {
      card: {
        brand: paymentMethod.card.brand,
        last4: paymentMethod.card.last4,
        exp_month: paymentMethod.card.exp_month,
        exp_year: paymentMethod.card.exp_year
      }
    }
  };
};

const mapDispatchToProps = (dispatch: any, ownProps: any) => {
  return {
    updateStripePaymentMethod: (data: any) => dispatch(updateStripePaymentMethod(ownProps.paymentMethodId, data.card.exp_month, data.card.exp_year)).then(() => dispatch(loadStripePaymentMethods()))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Update);
