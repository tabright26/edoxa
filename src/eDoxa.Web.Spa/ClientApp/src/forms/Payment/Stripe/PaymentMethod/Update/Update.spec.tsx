import React from "react";
import Update from "./Update";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

it("renders correctly", () => {
  const paymentMethod = {
    card: {
      brand: "visa",
      last4: "42w42",
      exp_year: 2030
    }
  };
  const store: any = {
    getState: () => {},
    dispatch: action => {},
    subscribe: () => {}
  };
  const tree = renderer
    .create(
      <Provider store={store}>
        <Update initialValues={paymentMethod} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
