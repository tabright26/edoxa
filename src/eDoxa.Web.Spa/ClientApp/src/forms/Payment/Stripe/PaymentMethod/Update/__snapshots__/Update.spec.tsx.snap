// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="d-flex form-inline"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="d-flex"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cc-visa fa-w-18 fa-2x "
        data-icon="cc-visa"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span
        className="my-auto ml-2"
      >
        XXXX XXXX XXXX 42w42
      </span>
    </div>
  </div>
  <div
    className="mx-auto"
  >
    <label
      className="ml-4 mr-2 text-muted"
    >
      Expiration:
    </label>
    <select
      className="d-inline form-control-sm form-control"
      disabled={false}
      name="card.exp_month"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "width": "55px",
        }
      }
      value=""
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
      <option
        value={6}
      >
        6
      </option>
      <option
        value={7}
      >
        7
      </option>
      <option
        value={8}
      >
        8
      </option>
      <option
        value={9}
      >
        9
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={11}
      >
        11
      </option>
      <option
        value={12}
      >
        12
      </option>
    </select>
    <span
      className="d-inline mx-2"
    >
      /
    </span>
    <select
      className="d-inline form-control-sm form-control"
      disabled={false}
      name="card.exp_year"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "width": "55px",
        }
      }
      value=""
    >
      <option
        value={2030}
      >
        2030
      </option>
      <option
        value={2031}
      >
        2031
      </option>
      <option
        value={2032}
      >
        2032
      </option>
      <option
        value={2033}
      >
        2033
      </option>
      <option
        value={2034}
      >
        2034
      </option>
      <option
        value={2035}
      >
        2035
      </option>
      <option
        value={2036}
      >
        2036
      </option>
      <option
        value={2037}
      >
        2037
      </option>
      <option
        value={2038}
      >
        2038
      </option>
      <option
        value={2039}
      >
        2039
      </option>
      <option
        value={2040}
      >
        2040
      </option>
      <option
        value={2041}
      >
        2041
      </option>
      <option
        value={2042}
      >
        2042
      </option>
      <option
        value={2043}
      >
        2043
      </option>
      <option
        value={2044}
      >
        2044
      </option>
      <option
        value={2045}
      >
        2045
      </option>
      <option
        value={2046}
      >
        2046
      </option>
      <option
        value={2047}
      >
        2047
      </option>
      <option
        value={2048}
      >
        2048
      </option>
      <option
        value={2049}
      >
        2049
      </option>
      <option
        value={2050}
      >
        2050
      </option>
    </select>
  </div>
  <button
    aria-label={null}
    className="btn btn-primary btn-sm"
    onClick={[Function]}
    style={
      Object {
        "width": "75px",
      }
    }
    type="submit"
  >
    Save
  </button>
</form>
`;
